core/math/geometry.cpp

```diff
@@ -913,8 +913,8 @@ Vector<Plane> Geometry::build_sphere_planes(real_t p_radius, int p_lats, int p_l
 			// FIXME: This is stupid.
 			Vector3 angle = normal.linear_interpolate(axis, j / (real_t)p_lats).normalized();
 			Vector3 pos = angle * p_radius;
-			planes.push_back(Plane(pos, angle));
-			planes.push_back(Plane(pos * axis_neg, angle * axis_neg));
+			planes.push_back(Plane(angle, pos));
+			planes.push_back(Plane(angle * axis_neg, pos * axis_neg));
```

Comment:
We can actually just use the `Plane::Plane(const Vector3 &p_normal, real_t p_d)` constructor here (and then we can remove the `// FIXME: This is stupid.` comment because it wouldn't be stupid anymore). It should be `Plane(angle, p_radius)` for the first one, for the second I'm not sure but I think it would be `Plane(angle * axis_neg, p_radius)`.

Reply:
Sorry, was without a PC for a while - adjusted everything accordingly.

Reply:
I'll take a closer look at the sphere planes later, just noticed the test case failed.